src/features/positions/stores/transform/index.ts

```diff
@@ -0,0 +1,512 @@
+      rainbowPosition.chainIds.push(position.chainId);
+      rainbowPosition.chainIds.sort((a, b) => a - b);
+    }
+
+    if (position.protocolVersion && !rainbowPosition.protocol_version) {
+      rainbowPosition.protocol_version = position.protocolVersion;
+    }
+
+    if (position.dapp && (!rainbowPosition.dapp.icon_url || position.dapp.iconUrl)) {
+      rainbowPosition.dapp = {
+        name: normalizeDappName(position.dapp.name || canonicalName),
+        url: position.dapp.url || '',
+        icon_url: position.dapp.iconUrl || '',
+        colors: position.dapp.colors || { primary: '#000000', fallback: '#808080', shadow: '#000000' },
+      };
+    }
+
+    position.portfolioItems.forEach(item => {
+      if (shouldFilterPositionType(item.name)) {
+        return;
+      }
+
+      const categories = mapPortfolioItemsToCategories(item);
+      const processed = transformCategories(categories, item, position, currency);
+
+      rainbowPosition.deposits.push(...processed.deposits);
+      rainbowPosition.pools.push(...processed.pools);
+      rainbowPosition.stakes.push(...processed.stakes);
+      rainbowPosition.borrows.push(...processed.borrows);
+      rainbowPosition.rewards.push(...processed.rewards);
+    });
+  });
+
+  return grouped;
+}
+
+// ============ Main Transform ================================================= //
+
+/**
+ * Transform DeBank API response into Rainbow positions format
+ * Groups by protocol, calculates totals, applies filters, and sorts by value
+ */
+export function transformPositions(response: ListPositionsResponse, params: PositionsParams): RainbowPositions {
+  const { currency } = params;
+  const emptyDisplay = convertAmountToNativeDisplay('0', currency as NativeCurrencyKey);
+
+  if (!response?.result?.positions || response.result.positions.length === 0) {
+    return {
+      positions: {},
+      positionTokens: [],
+      totals: {
+        total: {
+          amount: '0',
+          display: emptyDisplay,
+        },
+        totalDeposits: {
+          amount: '0',
+          display: emptyDisplay,
+        },
+        totalBorrows: {
+          amount: '0',
+          display: emptyDisplay,
+        },
+        totalRewards: {
+          amount: '0',
+          display: emptyDisplay,
+        },
+        totalLocked: {
+          amount: '0',
+          display: emptyDisplay,
+        },
+      },
+    };
+  }
+
+  const grouped = groupByProtocol(response.result.positions, currency);
+
+  Object.values(grouped).forEach(position => {
+    calculatePositionTotals(position, currency);
+  });
+
+  const filtered = filterPositions(grouped);
+  const sorted = sortPositions(filtered);
+
+  const positionsRecord = Object.fromEntries(sorted.map(position => [position.type, position]));
+  const grandTotals = calculateGrandTotals(sorted, currency);
+
+  return {
+    positions: positionsRecord,
+    positionTokens: response.result.uniqueTokens || [],
```

Comment:
Would reuse a static empty array here and anywhere else we use empty array fallbacks for store data (where the array is empty and never written to), to preserve object identity between empty wallets.